docs/getting-started/overview.rst

```diff
@@ -1,6 +1,110 @@
+
+The .NET Framework is the premier implemetentation of the .NET Platform available for 
+Windows server and client developers. It is a very powerful, very mature framework, with 
+a huge class library (known as the **Framework Class Library**) that supports 
+a wide variety of applications and solutions on Windows. 
+
+The additional stacks built on top the .NET Framework allow developers to build sofware 
+in wide range from console applications to rich web applications. 
+
+`Windows Forms <https://msdn.microsoft.com/en-us/library/dd30h2yb(v=vs.110).aspx>`_ 
+and `Windows Presentation Foundation (WPF) <https://msdn.microsoft.com/en-us/library/ms754130(v=vs.110).aspx>`_ 
+are two User Interface (UI) stacks that allow you to build desktop applications for Windows 
+operating system. Windows Forms' strength is in its rich support for common databinding as well as 
+access to Windows' native user interface controls. WPF, on the other hand, allows you to exercise 
+much more control over the look and feel of your application. Both of them allow for building very 
+rich desktop applications that run on Windows, and you should pick the one that is suited for your 
+use case. 
+
+`Windows Communication Foundation (WCF) <https://msdn.microsoft.com/en-us/library/ms731082(v=vs.110).aspx>`_ 
+is a set of libraries that comprise the middleware services stack on .NET Framework. Its main goal 
+is to allow you to use one set of libraries to create services that can communicate through various 
+supported protocols using various data formats and to be hosted in any process you choose, so they 
+are not tied to any particular hosting strategy. WCF allows you to use both 
+`SOAP <https://en.wikipedia.org/wiki/SOAP>`_ and 
+`REST <https://en.wikipedia.org/wiki/Representational_state_transfr>`_ based approaches to building 
+services. 
+
+**ASP.NET** is the .NET web framework. Being a very rich framework, it has several disticnt pieces 
+which can all be used in producing modern and high-performance web applications. 
+`ASP.NET Web Forms <http://www.asp.net/web-forms>`_ is a set of tools geared primarily towards 
+developer productivity, allowing quick turnaround on web applications with a drag-and-drop surface 
+reusing web controls for everything from loging in to data binding. 
+`ASP.NET MVC <http://www.asp.net/mvc>`_ allows for a different approach, one that gives you greater 
+control over the entire pipeline, from HTTP layer to the user interface. 
+`ASP.NET WebAPI <http://www.asp.net/web-api>`_ is a convention-based framework for creating REST 
+services. It allows you to stand up a REST endpoint extremely fast. Finally, 
+`SignalR <http://www.asp.net/signalr>`_ allows you to provide push-based communication to your web 
+applications. 
+
+
+.NET Native
+-----------
+
+.NET Native is not so much an "edition" of the .NET platform as it is a set of tools that 
+allow developers to have different build outputs. The normal .NET compilation process takes 
```

Comment:
I would use the term "source compilation" and then would differentiate with "native compilation" or "IL byte code compilation" later.